### ISSUE TITLE
mpiext_cuda_c.h include in mpi_ext.h causing problems

### ISSUE THREAD

**Issue by hppritcha** [open-mpi/ompi#6069]
We're noticing problems like this in our MTT running master:

```
15024 stdout:   CC       allgather_init_inter.o
15025 stderr: In file included from allgather_init_inter.c:12:
15026 stderr: /Users/debrezanka/sandbox/mtt/mttscratch/MiddlewareBuild_OMPIMaster/include/mpi-ext.h:      20:10: fatal error: 'openmpi/ompi/mpiext/cuda/c/mpiext_cuda_c.h' file not found
15027 stderr: #include "openmpi/ompi/mpiext/cuda/c/mpiext_cuda_c.h"
15028 stderr:          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
15029 stderr: 1 error generated.
15030 stderr: make[2]: *** [allgather_init_inter.o] Error 1
15031 stderr: make[1]: *** [all-recursive] Error 1
15032 stderr: make: *** [all-recursive] Error 1
```
looks like its a combination of cuda support not being asked for in our builds and the way configury builds mpi_ext.h.

**Comment by Akshay-Venkatesh** [comment]
Hi Howard. This could be an error in the test.

> We're noticing problems like this in our MTT running master:
> 
> ```
> 15024 stdout:   CC       allgather_init_inter.o
> 15025 stderr: In file included from allgather_init_inter.c:12:
> 15026 stderr: /Users/debrezanka/sandbox/mtt/mttscratch/MiddlewareBuild_OMPIMaster/include/mpi-ext.h:      20:10: fatal error: 'openmpi/ompi/mpiext/cuda/c/mpiext_cuda_c.h' file not found
> 15027 stderr: #include "openmpi/ompi/mpiext/cuda/c/mpiext_cuda_c.h"
> 15028 stderr:          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

included file here looks hard-coded.  My guess is that it should just be `ompi/mpiext/cuda/c/mpiext_cuda_c.h` and it should have some under some ifdef guard (included when when cuda support is requested).

> 15029 stderr: 1 error generated.
> 15030 stderr: make[2]: *** [allgather_init_inter.o] Error 1
> 15031 stderr: make[1]: *** [all-recursive] Error 1
> 15032 stderr: make: *** [all-recursive] Error 1
> ```
> looks like its a combination of cuda support not being asked for in our builds and the way configury builds mpi_ext.h.

**Comment by hppritcha** [comment]
I don't think your analysis is accurate.  allgather_init_inter is just pulling in our mpi-ext.h header file
for the persistent collectives extensions.

**Comment by hppritcha** [comment]
cisco tests are also showing this failure
@jsquyres

**Comment by kawashima-fj** [comment]
@ggouaillardet Is this related to f8318f0a8ffa9d0ca109a7fbb6ebec9f5522a687?

**Comment by ggouaillardet** [comment]
@hppritcha `mpi-ext.h` pulls __all__ the extensions headers, so `mpiext_cuda_c.h` should be pulled if CUDA was found at configure time.

@kawashima-fj let me take a look at it, it could be a side effect of the mentioned commit after all ...

**Comment by ggouaillardet** [comment]
Sorry for that, this (and my initial issue) should be fixed now in #6071
